--- a/config/gsad.service.in
+++ b/config/gsad.service.in
@@ -7,12 +7,26 @@
 [Service]
 Type=exec
 User=gvm
+Group=gvm
 PIDFile=${GSAD_PID_PATH}
-RuntimeDirectory=gsad
+RuntimeDirectory=gvm
 RuntimeDirectoryMode=2775
-ExecStart=${SBINDIR}/gsad --foreground --listen 127.0.0.1 --port 9392 --http-only
+EnvironmentFile=${DEFAULT_CONFIG_DIR}/gsad
+ExecStart=${SBINDIR}/gsad --listen $GSAD_ADDRESS --port $GSAD_PORT $GSAD_OPTIONS
 Restart=always
 TimeoutStopSec=10
+# gsad service requires serverkey.pem created by gvmd
+# post-installation. When the user install the gvm package it
+# may configure after that. In this case systemd will try to
+# restart the gsad too fast and it will stop quickly with the
+# error "Start request repeated too quickly". Adding gvmd as a
+# PreDepends dependency may solve the issue but it make
+# difficulties for the docker installation which will require
+# separate containers for gsad and gvmd. So for now the best
+# solution it is to add a RestartSec to the systemd service to
+# make more time for systemd service to wait for gvmd to
+# configure.
+RestartSec=3s
 
 [Install]
 WantedBy=multi-user.target
